File: JS/Airport.js
import PassengerPlane from './Planes/PassengerPlane.js';
import MilitaryPlane from './Planes/MilitaryPlane.js';
import militaryType from './models/militaryType.js';
import ExperimentalPlane from './Planes/ExperimentalPlane.js';

class Airport {
  static print(planes) {
    return JSON.stringify(planes);
  }

  constructor(planes) {
    this.planes = planes;
  }

  getPlanes() {
    return this.planes;
  }

  getPassengerPlanes() {
    return this.planes.filter((plane) => plane instanceof PassengerPlane);
  }

  getMilitaryPlanes() {
    return this.planes.filter((plane) => plane instanceof MilitaryPlane);
  }

  getExperimentalPlanes() {
    return this.planes.filter((plane) => plane instanceof ExperimentalPlane);
  }

  getTransportMilitaryPlanes() {
    return this.getMilitaryPlanes().filter((plane) => plane.getMilitaryType() === militaryType.TRANSPORT);
  }

  getBomberMilitaryPlanes() {
    return this.getMilitaryPlanes().filter((plane) => plane.getMilitaryType() === militaryType.BOMBER);
  }

  getPassengerPlaneWithMaximumPassengersCapacity() {
    return this.getPassengerPlanes()
      .sort((a, b) => b.getMaximumPassengersCapacity() - a.getMaximumPassengersCapacity())[0];
  }

  sortByMaximumDistance() {
    this.planes = this.planes
      .sort((a, b) => b.getMaximumFlightDistance() > a.getMaximumFlightDistance());
    return this;
  }

  sortByMaximumSpeed() {
    this.planes = this.planes.sort((a, b) => b.getMaximumSpeed() - a.getMaximumSpeed());
    return this;
  }

  sortByMaximumLoadCapacity() {
    this.planes.sort((a, b) => b.getMaximumLoadCapacity() > a.getMaximumLoadCapacity());
    return this;
  }
}

export default Airport;

File: Net/Aircompany/Airport.cs
using System.Collections.Generic;
using System.Linq;
using Aircompany.Models;
using Aircompany.Planes;

namespace Aircompany
{
    public class Airport
    {
        public List<Plane> Planes;

        public Airport(IEnumerable<Plane> planes)
        {
            Planes = planes.ToList();
        }

        public IEnumerable<PassengerPlane> GetPassengersPlanes() => _planes.OfType<PassengerPlane>();

        public IEnumerable<MilitaryPlane> GetMilitaryPlanes() => Planes.OfType<MilitaryPlane>();

        public PassengerPlane GetPassengerPlaneWithMaximumPassengersCapacity()
        {
            var passengerPlanes = GetPassengersPlanes();
            return passengerPlanes.Aggregate((firstPlane, secondPlane) => firstPlane.PassengersCapacity > secondPlane.PassengersCapacity ? firstPlane : secondPlane);
        }

        public List<MilitaryPlane> GetTransportMilitaryPlanes()
        {
            var militaryPlanesList = GetMilitaryPlanes();
            var type = "Transport";
            return militaryPlanesList.Where(plane => plane.mililatyType == type);
        }

        public Airport SortByMaximumDistance() => new Airport(Planes.OrderByDescending(plane => plane.MaxFlightDistance));

        public Airport SortByMaximumSpeed() => new Airport(Planes.OrderByDescending(plane => plane.MaxSpeed));

        public Airport SortByMaximumLoadCapacity() => new Airport(Planes.OrderByDescending(plane => plane.MaxLoadCapacity));

        public IEnumerable<Plane> GetPlanes() => Planes;

        public override string ToString()
        {
            return "Airport: planes=" + string.Join(", ", Planes.Select(plane => plane.Model)) + ".";
        }
    }
}
